test/test_util/custom_matchers.dart

```diff
@@ -55,7 +55,9 @@ class ClassNameMatcher extends Matcher {
   }
 
   @override
-  bool matches(String className, Map matchState) {
+  bool matches(className, Map matchState) {
+    className = className as String; // ignore: avoid_as
```

Comment:
We wouldn't have to do this if this were to be accepted https://github.com/dart-lang/matcher/issues/37